salt/states/file.py

```diff
@@ -929,10 +929,10 @@ def symlink(
                     return _error(ret, ((
                                             'File exists where the backup target {0} should go'
                                         ).format(backupname)))
-                elif os.path.isfile(backupname):
-                    os.remove(backupname)
-                elif os.path.isdir(backupname):
-                    shutil.rmtree(backupname)
+                elif os.path.isfile(backupname)\
```

Comment:
The original logic makes sense because different functions in the Python stdlib need to be called based on what needs to be deleted. (i.e., os.remove() will raise `OSError` when trying to remove a directory). However, this logic is all handled internally in salt's `file.remove` so I don't see a need to replicate it here.